app/core/src/main/java/com/fsck/k9/controller/push/AlarmPermissionManagerApi21.kt

```diff
@@ -0,0 +1,14 @@
+package com.fsck.k9.controller.push
+
+/**
+ * On Android versions prior to 12 there's no permission to limit an app's ability to schedule exact alarms.
+ */
+internal class AlarmPermissionManagerApi21 : AlarmPermissionManager {
+    override fun canScheduleExactAlarms(): Boolean {
+        return true
```

Comment:
This is basically the same as `AlarmManagerCompat.canScheduleExactAlarms()` but now we have that logic in two places.

Reply:
I don't think `androidx.core.app.AlarmManagerCompat` will add support for listening to the broadcast intent. So we'll still need our own "compat" variant of that. Injecting `AlarmManager` and `AlarmManagerCompat` here to be able to call [`canScheduleExactAlarms()`](https://developer.android.com/reference/androidx/core/app/AlarmManagerCompat#canScheduleExactAlarms(android.app.AlarmManager)) when we know it will return `true` feels unnecessary.

We could separate `canScheduleExactAlarms()` from `registerListener()` and `unregisterListener()`. But that would mainly lead to more lines of code and files, not any tangible benefit.

Reply:
We could change the naming, but it's not super important.